packages/editor/src/browser/editor-command.ts

```diff
@@ -163,6 +166,15 @@ export class EditorCommandContribution implements CommandContribution {
     @inject(ResourceProvider)
     protected readonly resourceProvider: ResourceProvider;
 
+    @postConstruct()
+    protected init(): void {
+        this.editorPreferences.onPreferenceChanged(e => {
+            if (e.preferenceName === 'editor.autoSave' && e.newValue === 'on') {
+                this.shell.saveAll();
```

Comment:
Sorry for hijacking this PR, I am sure this solution works flawlessly but this is a generic ~issue~ inconveniency; almost every time when I update to `next` Theia in a downstream project, I have to resolve a cycle in the DI as we just inject services here and there.
Instead of injecting the `shell`, can we do the following instead?
```ts
this.commandService.executeCommand(CommonCommands.SAVE_ALL.id);
```
Or make the editor command contribution a frontend app contribs, and save a reference to the app, then you can get the shell.

Long story short, injecting thing here, and they usually cause a DI cycle after a Theia update. Can we make the modules less dependent on each other?

Reply:
Sorry @kittaakos, I was unaware that using injections like I've done causes issues, even if `ApplicationShell` is from `@theia/core`. I can make the changes like you suggested, should we add a note in the coding guidelines (I was not aware about the DI cycle issues)?

Reply:
> Sorry @kittaakos, I was unaware that using injections like I've done causes issues, even if `ApplicationShell` is from `@theia/core`. I can make the changes like you suggested, should we add a note in the coding guidelines (I was not aware about the DI cycle issues)?

No worries at all. I think we can leave this PR as is, but it would be nice to have a strategy for the future as the number of possible DI cycles increases in downstream projects when the `Application`, `ApplicationShell`, `ShellLayoutRestorer`, etc are all customized.

Reply:
In order to use `commandService` (for execute command) I'd need to inject it anyways.
Is there a preference between injecting the `ApplicationShell` versus `CommandService`?

Reply:
`CommandService` does not have injected members (fields). So it won't cause a DI cycle.

Reply:
> `CommandService` does not have injected members (fields). So it won't cause a DI cycle.

Thank you for clarifying for me, I took the liberty to update the pull-request without the use of any new injections :+1:

Reply:
Could you revert latest changes please? Calling `save` on the shell is important.